### PR TITLE
Use necojackarc/auto-request-review action to request reviews from non-maintainers

### PR DESCRIPTION
**Pull request by sunjayBhatia**
CODEOWNERS only allows requesting reviews from users/teams with owner permission on the repo

Use this action which should not require the reviewers github team to have owner permissions

### CODE REVIEW

**Comment by sunjayBhatia** [comment]
As part of this we might need to disable reviews on the @projectcontour/maintainers team to let this action handle all the reviewer logic

**Comment by sunjayBhatia** [comment]
Also may want to unset this: https://github.com/projectcontour/contour/blob/6917d85ce41f3050107994c06e6bb5b7039c99ed/.github/dependabot.yml#L17

**Comment by sunjayBhatia** [comment]
unfortunately to get access to the secret needed to access teams, had to switch to the `pull_request_target` event to trigger the workflow on, which then won't run on this PR since the base branch doesn't include the workflow

**Comment by sunjayBhatia** [comment]
we'll likely have to merge this w/o knowing if it works so that PRs have access to the secret/etc. to run this action

File: .github/reviewers.yaml
Comment (skriss) [inline .github/reviewers.yaml:8]: Is there any way to ensure that at least one maintainer is selected?

Comment (sunjayBhatia) [reply .github/reviewers.yaml:8]: ah yeah there might be a way to do so with the "groups" feature, but it might be simpler to keep the CODEOWNERS and just specify reviewers with this action

Comment (skriss) [reply .github/reviewers.yaml:8]: Circling back here, I'm in favor of keeping CODEOWNERS for existing maintainers, and using this workflow for assigning non-maintainer reviewers.

**Comment by sunjayBhatia** [comment]
I'll need to regenerate the secret referenced in the workflow as I believe that has expired

**Review by skriss — APPROVED** [review]

**Comment by sunjayBhatia** [comment]
Ok regenerated token and merging

**Pull request merged**
